tools/generate_docker_image_names.py

```diff
@@ -0,0 +1,112 @@
+"""
+Dockerリポジトリ名、バージョン文字列、カンマ区切りのプレフィックスを受け取り、
+バージョン文字列付きのDockerイメージ名を改行区切りで標準出力に出力する
+
+例1
+$ python3 ./tools/generate_docker_image_names.py \
+  --repository "REPO" \
+  --version "VER" \
+  --prefix ",A,B"
+REPO:VER
+REPO:A-VER
+REPO:B-VER
+
+例2
+$ python3 ./tools/generate_docker_image_names.py \
+  --repository "REPO" \
+  --version "VER" \
+  --prefix ""
+REPO:VER
+"""
+
+from argparse import ArgumentParser
+
+
+def generate_docker_image_names(
+    repository: str, version: str, comma_separated_prefix: str
+) -> list[str]:
+    """
+    Dockerリポジトリ名、バージョン文字列、カンマ区切りのプレフィックスを受け取り、
+    バージョン文字列付きのDockerイメージ名を配列で返す
+
+    prefixが空文字列でない場合、"{prefix}-{version}"をタグにする
+
+    - 例: repository="REPO", version="VER", prefix="A" -> "REPO:A-VER"
+
+    prefixが空文字列の場合、"{version}"をタグにする
+
+    - 例: repository="REPO", version="VER",  prefix="" -> "REPO:VER"
+
+    Parameters
+    ----------
+    repository : str
+        Dockerリポジトリ名（例: "voicevox/voicevox_engine"）
+    version : str
+        バージョン文字列（例: "0.22.0", "latest"）
+    comma_separated_prefix : str
+        カンマ区切りのプレフィックス（例: ",cpu,cpu-ubuntu22.04", "nvidia,nvidia-ubuntu22.04"）
+
+    Returns
+    -------
+    list[str]
+        Dockerイメージ名の配列
+        （例: [
+            "voicevox/voicevox_engine:0.22.0",
+            "voicevox/voicevox_engine:cpu-0.22.0",
+            "voicevox/voicevox_engine:cpu-ubuntu22.04-0.22.0"
+        ]）
+    """
```

Comment:
ちょっと記法が独特ですね！
関数のドキュメントに引数の例を書くのはあまりない気がしますね。受け付けられる値の一覧を書くとかなら見かけますが。

numpydoc形式には`Examples`項目を作れるらしいので、そっちにまとめるとかどうでしょう？
こうとか･･･？
```suggestion
    Parameters
    ----------
    repository : str
        Dockerリポジトリ名
    version : str
        バージョン文字列
    comma_separated_prefix : str
        カンマ区切りのプレフィックス

    Returns
    -------
    list[str]
        Dockerイメージ名の配列。

    Examples
    --------
    >>> generate_docker_image_names("voicevox/voicevox_engine", "0.22.0", "cpu,cpu-ubuntu22.04")
    ['voicevox/voicevox_engine:0.22.0',
     'voicevox/voicevox_engine:cpu-0.22.0',
     'voicevox/voicevox_engine:cpu-ubuntu22.04-0.22.0']
    """
```

Reply:
何も書かなくてもいい気もしてきましたが、書くならExamplesの方がよさそうですね！